src/render/md.jl

```diff
@@ -43,50 +56,84 @@ function writemd(io::IO, doc::Metadata, config::Config)
     end
 
     if !isempty(index)
-        ents = Entries()
-        for k in CATEGORY_ORDER
+        ents = prepare_entriesmd(index, doc, config)
+        if (has_items(ents.exported) || has_items(ents.internal))
+            config.md_genindex && (push!(API_INDEX.sourcepaths, abspath(filepath));
+                                   push!(API_INDEX.modulenames, string(modulename(doc)));
+                                   push!(API_INDEX.entriesmd, ents))
+            println(io)
+            process_entriesmd(io, "Exported", ents.exported, config)
+            process_entriesmd(io, "Internal", ents.internal, config)
+        end
+    end
+end
+
+function prepare_entriesmd(index::Dict{Symbol, Any}, doc::Metadata, config::Config)
+    ents = EntriesMD(config)
+    if (config.md_genindex || config.md_permalink)
+        pageanchors = Dict{Symbol, Dict{String, Int}}([])
+        for k in config.category_order
             haskey(index, k) || continue
+            k in pageanchors || (pageanchors[k] = Dict([]))
+            basenames = pageanchors[k]
             for (s, obj) in index[k]
-                push!(ents, modulename(doc), obj, entries(doc)[obj])
+                ent = entries(doc)[obj]
+                if k == :comment
+                    basename = "comment"
+                else
+                    basename = k == :macro                                          ?
+                            Base.lowercase(string(Docile.Interface.macroname(ent))) :
+                            Base.lowercase(string(Docile.Interface.name(obj)))
```

Comment:
```julia
basename = lowercase(string(k == :macro ? Interface.macroname(ent))) : Interface.name(obj)))
```

Add `import Docile.Interface` in the imports list if it's not already there. `lowercase` is an exported method as far as I recall.